cppapi/server/fwdattrdesc.cpp

```diff
@@ -180,7 +181,13 @@ bool FwdAttr::validate_fwd_att(vector<AttrProperty> &prop_list,const string &dev
 	}
 	catch (...) {}
 
-	if (root_att_db_defined == true)
+	//check if full_root_att is already set
+	if (full_root_att.size()!=0 && full_root_att.compare(RootAttNotDef)!=0)
```

Comment:
It does exactly the same -- I mean there is no `else` -- and much easier to read:

Instead of

```c++
if(is_full_root_att_set)
		; // root attribute was defined in constructor: do not overwrite it
else if (root_att_db_defined)
        full_root_att = root_att_db;
```

do

```c++
if(!is_full_root_att_set && root_att_db_defined)
    full_root_att = root_att_db;
```

 Because when you quickly look through the code, longer code requires much more efforts to understand what is going on. Believe me ;)